src/hera/shared/_global_config.py

```diff
@@ -201,13 +201,15 @@ def _set_defaults(cls, values):
 
 _SCRIPT_PYDANTIC_IO_FLAG = "script_pydantic_io"
 _DECORATOR_SYNTAX_FLAG = "decorator_syntax"
+_ARTIFACT_PATH_AS_STRING_FLAG = "artifact_path_as_string"
 
 # A dictionary where each key is a flag that has a list of flags which supersede it, hence
 # the given flag key can also be switched on by any of the flags in the list. Using simple flat lists
 # for now, otherwise with many superseding flags we may want to have a recursive structure.
 _SUPERSEDING_FLAGS: Dict[str, List] = {
     _SCRIPT_PYDANTIC_IO_FLAG: [_DECORATOR_SYNTAX_FLAG],
     _DECORATOR_SYNTAX_FLAG: [],
+    _ARTIFACT_PATH_AS_STRING_FLAG: [],
```

Comment:
would it be simpler to do `_SUPERSEDING_FLAGS.get(flag, [])` so we don't need to add empty lists here?